tests/unit/test_index.py

```diff
@@ -153,4 +153,4 @@ def test_get_formatted_locations_basic_auth():
     finder = PackageFinder([], index_urls, session=[])
 
     result = finder.get_formatted_locations()
-    assert 'user' not in result and 'pass' not in result
+    assert 'user' in result and 'pass' not in result
```

Comment:
How about also checking that “****” is in the result to check / make it more obvious that redaction should be happening?